services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/NeutronRouterServiceImpl.java

```diff
@@ -558,14 +568,30 @@ public UpdateRoutingRuleResponse updateRoutingRule (String owner, NewRoutesWebRe
             int newBitmaskInt = Integer.parseInt(newBitmask);
             // TODO: 2. flag check if it is from vpc/Neutron(lower priority) or subnet.
             RouteEntry existRoute = existRoutesMap.get(newNetworkIP);
-            // dont find: Create (both)
+            // can't find: Add Operation - Create (both); Remove Operation - Skip
             if (existRoute == null) {
+
+                if (!isAddOperation) { // Remove Operation - Skip
+                    continue;
+                }
```

Comment:
@kevin-zhonghao If its delete operation, do we need to identify which route need to be deleted? If not, why don't we just insert each route in the routetable into `internalsubnetroutingtable` with *delete* operation.

Reply:
There is a prerequisite that is new routes cannot be found in the existing route table.
 If its delete operation and we cannot found it in the existing route table, that means we don't delete it because we just the routes exist in route table, right?

Reply:
@kevin-zhonghao You are right. For the current design, we still need to match the routes to be deleted, since we are rely on the entire route table from user's input to update the existing rules. In the future, we better based on each induvial rule to update rules.